cmd/dep/gopath_scanner.go

```diff
@@ -53,8 +53,9 @@ func (g *gopathScanner) InitializeRootManifestAndLock(rootM *dep.Manifest, rootL
 	}
 
 	g.origM = &dep.Manifest{
-		Constraints: g.pd.constraints,
-		Ovr:         make(gps.ProjectConstraints),
+		Constraints:  g.pd.constraints,
+		Ovr:          make(gps.ProjectConstraints),
+		PruneOptions: gps.PruneNestedVendorDirs,
```

Comment:
I was basing this on the assumption that pruning is opt-in. So we only remove nested vendor directories by default.